### PR TITLE
Simplify HostAnalysisScope storage

### PR DESCRIPTION
**Pull request by sharwell**
Starting with #28918, this type only ever exists in the context of a single analyzer, so we can remove the maps it contains and simplify calls into it.

### CODE REVIEW

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs [hunk 1]
```diff
@@ -433,11 +427,9 @@ public override AnalyzerActions GetAnalyzerActions(DiagnosticAnalyzer analyzer)
     /// <summary>
     /// Scope for setting up analyzers for analyzing a symbol and its members.
     /// </summary>
-    internal sealed class HostSymbolStartAnalysisScope : HostAnalysisScope
+    internal sealed class HostSymbolStartAnalysisScope(DiagnosticAnalyzer analyzer)
```
Comment (ToddGrun) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:430]: >HostSymbolStartAnalysisScope [](http://example.com/codeflow?start=26&length=28)

Should HostCodeBlockStartAnalysisScope and HostOperationBlockStartAnalysisScope also hold onto an analyzer so their Register* methods need not take in an analyzer for consistency?

Comment (ToddGrun) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:430]: Which would make AnalyzerCodeBlockStartAnalysisContext and AnalyzerOperationBlockStartAnalysisContext not need to take in an analyzer directly, right?

Comment (sharwell) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:430]: ⏱️ I will look into this as soon as https://github.com/dotnet/arcade/issues/14761 is resolved. Until that time I can't really make progress in an arcade-enabled repository due to a wall of errors.

Comment (ToddGrun) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:430]: Wanted to ping on this PR, as I think it's a good change and would like to see it move forward. Are you still blocked by the Arcade issue?

Comment (sharwell) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:426]: ➡️ These types are now updated for consistency

**Review by ToddGrun — APPROVED** [review]

**Comment by sharwell** [comment]
@dotnet/roslyn-compiler for reviews

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs [hunk 2]
```diff
@@ -329,10 +320,9 @@ public override void RegisterSyntaxNodeAction(Action<SyntaxNodeAnalysisContext>
     /// </summary>
     internal sealed class AnalyzerOperationBlockStartAnalysisContext : OperationBlockStartAnalysisContext
     {
-        private readonly DiagnosticAnalyzer _analyzer;
         private readonly HostOperationBlockStartAnalysisScope _scope;
 
-        internal AnalyzerOperationBlockStartAnalysisContext(DiagnosticAnalyzer analyzer,
+        internal AnalyzerOperationBlockStartAnalysisContext(
                                                             HostOperationBlockStartAnalysisScope scope,
```
Comment (jjonescz) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:326]: Nit: a bit weird formatting (applies to other similar places)


```suggestion
        internal AnalyzerOperationBlockStartAnalysisContext(HostOperationBlockStartAnalysisScope scope,
```

**Review by jjonescz — APPROVED** [review]

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs (hunk 2 above)
Comment (sharwell) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:326]: ➡️ I intentionally left formatting as-is to reduce the diff size. The outcome would have looked normal if the original code had adhered to better formatting practices, but it wasn't my place to alter it.

**Comment by sharwell** [comment]
@dotnet/roslyn-compiler for second review

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs [hunk 3]
```diff
@@ -19,102 +19,100 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     internal sealed class AnalyzerAnalysisContext : AnalysisContext
```
Comment (333fred) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs:20]: Man, this type itself is starting to feel a bit redundant now...

**Review by 333fred — APPROVED** [review]

**Pull request merged**